guide/src/reference/attributes/on-js-imports/host_binding.md

```diff
@@ -0,0 +1,139 @@
+# `host_binding`
+
+The `host_binding` attribute is the converse of the [`structural`
+attribute](structural.html). It configures how `wasm-bindgen` will generate JS
+glue to call the imported function. The naming here is intended convey that this
+attribute is intended to implement the semantics of the future [host bindings
+proposal][host-bindings] for WebAssembly.
+
+[host-bindings]: https://github.com/WebAssembly/host-bindings
+[reference-types]: https://github.com/WebAssembly/reference-types
+
+The `host_binding` attribute is intended to be purely related to performance. It
+ideally has no user-visible effect, and well-typed `structural` imports (the
+default) should be able to transparently switch to `host_binding` eventually.
+
+The eventual performance aspect is that with the [host bindings
```

Comment:
I think this is also misleading, since `structural` is generally faster, and the amount of JS shims is the same (see my comment below comparing `structural` and `host_binding`).

The only potential performance difference is whether it needs to lookup the prototype chain once (with `host_binding`) or multiple times (with `structural`).

Reply:
I think this is appropriately worded as-is because it explicitly says the "eventual performance", not commenting on today's performance. This is quickly becoming an incredibly niche feature that basically no one will use, so I don't think it's necessary to have absolutely exhaustive documentation for such a feature which goes into 100% of the nuances of why one might or might not want to use the attribute